### PR TITLE
Cleanup hlo_extractor and hlo_bisect dependecies

### PR DESCRIPTION
**Pull request by shraiysh**
//xla/tools:hlo_extractor and //xla/tools/hlo_bisect:hlo_bisect_state depend on //xla/tests:test_utils. This dependency should not exist. Moving the relevant functions to //xla:literal_util.

### CODE REVIEW

File: xla/tools/hlo_extractor.cc [hunk 1]
```diff
@@ -42,7 +42,6 @@ limitations under the License.
 #include "xla/service/hlo_verifier.h"
 #include "xla/shape.h"
 #include "xla/shape_util.h"
-#include "xla/tests/test_utils.h"
```
Comment (akuegel) [inline xla/tools/hlo_extractor.cc:45]: Please add xla/literal_util.h include instead. Also the corresponding BUILD dependency.

**Review by akuegel — APPROVED** [review]
Thanks

File: xla/tools/hlo_extractor.cc (hunk 1 above)
Comment (shraiysh) [reply xla/tools/hlo_extractor.cc:45]: That is already included in line number 38 and the dependency also exists already.

Comment (shraiysh) [reply xla/tools/hlo_extractor.cc:45]: It is already added in line 39. And the dependency also exists.

Comment (akuegel) [reply xla/tools/hlo_extractor.cc:45]: Ah, sorry, I should have expanded the view to see it.

**Review by akuegel — APPROVED** [review]

**Pull request closed without merging**